gateway/http.d/init.conf

```diff
@@ -15,7 +15,7 @@ init_by_lua_block {
     -- This ENV variable is defined in the main nginx.conf.liquid and injected when including this partial.
     -- The content of the ENV variable is a Lua table, so when rendered it actually can run ipairs on it.
     for k,v in pairs({{ ENV }}) do
-      if type(k) == 'string' and k ~= 'APICAST_CONFIGURATION_LOADER' then
+      if type(k) == 'string' and not resty_env.value(k) then
```

Comment:
I don't get this change. Can you explain it, please?

Reply:
It sets only missing/empty variables.

This leaves the flexibility of overriding the ENV when starting nginx and treating the `{{ ENV }}` only as default values.

Because Test::APIcast runs `APICAST_CONFIGURATION_LOADER=test bin/apicast` and expects it to work. But in cases where `APICAST_CONFIGURATION_LOADER=boot` is already set in the ENV this would not apply and `APICAST_CONFIGURATION_LOADER` would have value `boot` during initialization. Basically ignoring what ENV it was started in.

So this change is for two reasons:

* I think it makes sense for ENV of the shell to take priority than what is in the config file
* Test::APIcast needs to start APIcast with APICAST_CONFIGURATION_LOADER=test when testing configuration with `--test` but then later start it with `boot` when the config is started for real.